### ISSUE TITLE
[Task]: Git sync issues

### ISSUE THREAD

**Issue by rishabhsaxena** [appsmithorg/appsmith#9209]
Summary (can ignore other comments on the thread)
- [x] #9751
- [x] #9758
- [x] #9831
- [x] Disconnect git flow (in progress)
- [x] When the user edits the application name in sub-branch other than master that particular application name will not be updated (@abhvsn), repro steps: create a child branch; rename app; refresh the page; the name is reset
- [x] When user delete a page and try to merge it with a master the internal server error is displayed (@abhvsn)
- [x] When the branch is deleted from the Git the branch should be deleted from the application (@AnaghHegde)
- [x] Throw an error while trying to switch to a remote branch when the local tracking branch exists (@AnaghHegde)
 - [x] Under Merge tab the conflict error is displayed with The branch committed change which is contradicting error (@RakshaKShetty could not repro this)
 - [x] Show merge success state, similar to commit and push success: https://www.figma.com/file/dHbJ2KNXa9597W6BzsFIuE/Local-Dev-WIP?node-id=754%3A8621
- [x] Remove border for branch dropdown within the merge tab
- [x] Remove remote branches from the branch dropdown on the merge tab
- [x] Show count of uncommitted changes on the commit (+) button at the buttom bar similar to how we show it for pull button
- [x] Bottom bar icons (@rishabhsaxena)
  - [x] improve spacing between the icons
  - [x] the number badge looks a bit odd along with the icons
  - [x] update tooltip text (append changes): `commit and push changes`, `push changes`, `pull changes`
- [x] Remove redundant error and success toasts, if the success or the error state is already indicated on the UI or the success toast is distracting we should remove those (@rishabhsaxena)
- [x] Append (*) to the branch button, if we have uncommitted changes (@rishabhsaxena)
- [x] Display remote and local branches within the branch list:
  - [x] bottom bar (@rishabhsaxena)
  - [x] merge tab branch dropdown (@Debsourabh)
- [x] An icon on the application card to indicate the application is connected to git (@rishabhsaxena)
- [x] The view application and edit application link doesn't work in the readme file (@AnaghHegde)
- [x] Add a check for existing files within the repo, repos with files apart from common templates such as: `README.md`, `.gitignore`, `llicence` should be invalid (@AnaghHegde)
- [x] Update socket rooms to include branch for maintaining realtime cursors and concurrent editor warning (@rishabhsaxena )
- [x] Separate feature flag for git import (@rishabhsaxena )
- [x] Update generate key step UI: https://www.figma.com/file/dHbJ2KNXa9597W6BzsFIuE/Local-Dev-WIP?node-id=1%3A47090 (@haojin111 )
- [x] `0 page updated` should say `0 pages updated`, similarly for queries(@haojin111 )
- [x] Remove push button(@haojin111 )
- [x] After signing up as a new user, on opening the git sync modal we get an error toast, (update error handling to hide the error toast for `/profile/default` api), related to #9196(@haojin111 )
- [x] Add option to use global config
- [x] Add latest deploy preview time: didn't appear until the duration was in minutes(@haojin111 )
- [x] Link `open repo` button with repo link(@haojin111 )
- [x] Git connection tab: on error the window doesn't scroll all the way to the bottom(@haojin111 )
- [x] Hide latest deploy preview until the app is published for the first time(@haojin111 )
- [x] Update banner styles for error and warning: https://www.figma.com/file/dHbJ2KNXa9597W6BzsFIuE/Local-Dev-WIP?node-id=206%3A9285, https://www.figma.com/file/dHbJ2KNXa9597W6BzsFIuE/Local-Dev-WIP?node-id=684%3A8619(@haojin111 )
- [x] Use large buttons as CTAs(@haojin111 )
- [x] Use `_` as the branch name separator, instead of `-`(@haojin111 )
- [x] Merge form: remove icon(@haojin111 )
- [x] Merge CTA moves down when the status is rendered, add offset to prevent that(@haojin111 )
- [x] Remove the % from the progress bar(@haojin111 )
- [x] Fix alignment for the skeletons on the deploy tab(@haojin111 )
- [x] Merge tab: add max height to the branch list(@haojin111 )
- [x] Fetch `merge status` message from the API, instead of a hardcoded message for error and success states (@rishabhsaxena )
- [x] Show progress bar after clicking on `Merge` CTA as the loading state(@haojin111 )
- [x] Edge case issues for the git user config in the connect tab (both before connection and post connection)(@haojin111 )
- [x] Generate key button should be disabled until a valid URL in entered, also the button shifts when the error is rendered below the URL input(@haojin111 )
- [x] The error alignment in the merge tab needs to be fixed(@haojin111 )
- [x] The generate Key button should be displayed only after the Remote URL has been (@haojin111 )
- [x] when the user clicks on the push icon the deploy tab should open(@haojin111 )
- [x] when branch name is too big and user hover on it the tooltip is not displayed(@haojin111 )
- [x] The dropdown alignment displayed for the git connect should be polished(@haojin111 )
- [x] after generating the deploy key before clicking on connect close the pop-up and open it again the Remote URL is not visible in the input box(@haojin111 )
- [x] Under the merge tab when the user searches for the branch name which is not present the pop up closes all of a sudden(@haojin111 )
- [x] under merge tab when the branch name is too big the complete name is not displayed in the dropdown (@haojin111 )
- [x] When the new user gets added to the old repo when the other users were working the New user can do his first push this causes the conflict in the Other user's app (expected)
- [x] The merge button should be disabled when the User merge the changes from the github (not in scope for v1)
- [x] User should not be allowed to add 2 different application in the same git repo (duplicate)
- [x] when the user switches between the tab in the pop the loader is displayed in the Deployed tab (expected, fetches status)
- [x] There is a delay in displaying the  changes update (is it due to the loading state?) loading state for the fetch status call

**Comment by RakshaKShetty** [comment]
- [ ] when the user clicks on the pull icon the deploy tab should open
- [ ] when branch name is too big and user hover on it the tooltip is not displayed
- [ ] when the user switches between the tab in the pop the loader is displayed in the Deployed tab
- [ ] The dropdown alignment displayed for the git connect should be polished
- [ ] There is a delay in displaying the  changes update 
- [ ] Under the merge tab when the user searches for the branch name which is not present the pop up closes all of a sudden
- [ ] under merge tab when the branch name is too big the complete name is not displayed in the dropdown 
- [ ] When the user click on any icon from the left bottom corner the pull (down arrow) icon changes to a loader icon
- [ ] When the new user gets added to the old repo when the other users were working the New user can do his first push this causes the conflict in the Other user's app

**Comment by RakshaKShetty** [comment]
- [ ] after generating the deploy key before clicking on connect close the pop-up and open it again the Remote URL is not visible in the input box
- [ ] The error alignment in the merge tab needs to be fixed
